Add more potential matches for regex of input files

GiudGiud opened the pull request:
Attempts a bug fix for #17426 #19137

doc doesnt build locally (someone teach me maybe), so going to debug with this PR

GiudGiud committed:
Add more potential matches for regex of input files, closes #17426 #19137

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1030018/) on 584bde6 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/20898/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1030045/) on 584bde6 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>edbf9a</th>
      <th colspan=4>#20898 584bde</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>82.58%</td>
      <td>82.58%</td>
      <td>+0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>74737</td>
      <td>74739</td>
      <td>+2</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15765</td>
      <td>15763</td>
      <td>-2</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20898/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20898/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/20898/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/20898/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/20898/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/20898/coverage/contact)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/20898/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/20898/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/20898/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/20898/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/20898/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/20898/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/20898/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/20898/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/20898/coverage/navier_stokes)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/20898/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/20898/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/20898/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/20898/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/20898/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/20898/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/20898/coverage/richards)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/20898/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/20898/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/20898/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/20898/coverage/xfem)

</details>



This comment will be updated on new commits.

GiudGiud commented:
@cticenhour this should be ready for review.
As the new log shows, quite a few input file objects are not matched to a class, and therefore that class' documentation does not list these inputs. I think it's just a matter of being more consistent between the C++ object names and the input file names. Might require some renaming to fix. Not the scope for this PR imo

GiudGiud committed:
Adapt test to new functionaliy, which looks at source files instead of headers.
Limit output of unit test by adding the test folders for name discovery, refs #19137

GiudGiud commented:
umm only apps failures at this point
https://civet.inl.gov/job/1042583/

I think I ll need to do some patches to make it work, possibly break next

GiudGiud committed:
Add source argument to make sure inputs argument is properly parsed when building class database, refs #17426

GiudGiud commented:
@cticenhour this is ready!
so happy rn haha

cticenhour commented:
I've been seeing the intense force pushes - good job! I'll give a formal review this afternoon 😄

GiudGiud committed:
Better logic for AD objects documentation, refs #17426

GiudGiud commented:
seems I broke it haha
I ll look more. we want a lot of things from this

GiudGiud commented:
ok should be good now

cticenhour commented on python/MooseDocs/common/build_class_database.py:
```diff
@@ -106,6 +123,15 @@ def _match_child(objects, filename, match):
-        objects[key].inputs.add(filename)
+
+    # AD instances are caught for both the regular and AD object. This is done because:
+    # - in all cases I know of, the AD/nonAD are from the same template
```
Could we improve this logic by looking for and flagging dict objects post `_match_definition` that have the same matched header?

cticenhour replied:
Might be beyond the scope of this PR as you do successfully address the raised issues, and I cannot think of a case where there are AD/nonAD objects, templated or not, that have differing parameters and perform the same function.

GiudGiud replied:
sure can do, pushing a commit now

GiudGiud committed:
Address Casey's review, solidify AD/nonAD input file collection by verifying that they come from the same source file, refs #17426

cticenhour reviewed (approved):
Comment was addressed, and new class database now functions as expected for both separate and templated AD/nonAD pairs. Thanks for doing this!
